File: app/src/main/java/closer/vlllage/com/closer/api/models/GroupMessageResult.java
package closer.vlllage.com.closer.api.models;

import java.util.Date;
import java.util.List;

import closer.vlllage.com.closer.store.models.GroupMessage;
import closer.vlllage.com.closer.store.models.ReactionCount;

public class GroupMessageResult extends ModelResult {
    public String from;
    public String to;
    public String text;
    public String attachment;
    public Date created;
    public PhoneResult phone;
    public List<ReactionCount> reactions;


    public static GroupMessage from(GroupMessageResult result) {
        GroupMessage groupMessage = new GroupMessage();
        groupMessage.setId(result.id);
        groupMessage.setFrom(result.from);
        groupMessage.setTo(result.to);
        groupMessage.setText(result.text);
        groupMessage.setTime(result.created);
        groupMessage.setUpdated(result.updated);
        groupMessage.setAttachment(result.attachment);
        groupMessage.setReactions(result.reactions);
        return groupMessage;
    }
}

File: app/src/main/java/closer/vlllage/com/closer/api/models/GroupMessageResult.kt
package closer.vlllage.com.closer.api.models

import closer.vlllage.com.closer.store.models.GroupMessage
import closer.vlllage.com.closer.store.models.ReactionCount

class GroupMessageResult : ModelResult() {
    var from: String? = null
    var to: String? = null
    var text: String? = null
    var attachment: String? = null
    var phone: PhoneResult? = null
    var reactions: List<ReactionCount> = listOf()
    var replies: Int? = null

    companion object {
        fun from(result: GroupMessageResult): GroupMessage {
            val groupMessage = GroupMessage()
            groupMessage.id = result.id
            groupMessage.from = result.from
            groupMessage.to = result.to
            groupMessage.text = result.text
            groupMessage.time = result.created
            groupMessage.updated = result.updated
            groupMessage.attachment = result.attachment
            groupMessage.reactions = result.reactions
            groupMessage.replies = result.replies
            return groupMessage
        }
    }
}
